import.py

```diff
@@ -649,7 +652,63 @@ def write_to_file(cont: str, filename: str) -> None:
         f.write(cont)
 
 
+def prepend_file(filename: str, new_str: str) -> None:
+    with open(filename, "r") as original_f:
+        with open(filename + "_", "w") as new_f:
+            new_f.write(new_str)
+            new_f.write(original_f.read())
+    os.rename(filename + "_", filename)
+
+
+def download_decompme(parsed_url: urllib.parse.ParseResult) -> None:
+
+    try:
+        # unique decomp.me identifier for scratches
+        slug = parsed_url.path[1:].split("/")[1]
+
+        response_str = urllib.request.urlopen(f"https://decomp.me/api/scratch/{slug}")
+        response_json = json.load(response_str)
+        func_name = response_json["name"]
+        dirname = create_directory(func_name)
+        urllib.request.urlretrieve(
+            f"https://decomp.me/api/scratch/{slug}/export", f"{dirname}/export.zip"
+        )
+        with zipfile.ZipFile(f"{dirname}/export.zip", "r") as zip_ref:
+            zip_ref.extractall(dirname)
```

Comment:
would prefer to use a temporary file for the zip, or keep it in memory if `zipfile` allows

Reply:
I thought there is some value in keeping the zip as the original data, before making mods to it...
but I went ahead and made it keep it in memory like you asked